Build failure: ollama with CUDA

### Steps To Reproduce

Steps to reproduce the behavior:
```nix
services.ollama = {
  enable = true;
  acceleration = "cuda"
};
```
### Build log

```
ollama> -- Found CUDAToolkit: /nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/include (found version "12.2.140")
ollama> -- cuBLAS found
ollama> CMake Error at /nix/store/mbw6qprvn80p8k7z3lq44hnz6b332xf6-cmake-3.28.3/share/cmake-3.28/Modules/CMakeDetermineCompilerId.cmake:780 (message):
ollama>   Compiling the CUDA compiler identification source file
ollama>   "CMakeCUDACompilerId.cu" failed.
ollama>
ollama>   Compiler: /nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/nvcc
ollama>
ollama>
ollama>   Build flags:
ollama>
ollama>   Id flags: --keep;--keep-dir;tmp -v
ollama>
ollama>
ollama>
ollama>   The output was:
ollama>
ollama>   1
ollama>
ollama>   #$ _NVVM_BRANCH_=nvvm
ollama>
ollama>   #$ _SPACE_=
ollama>
ollama>   #$ _CUDART_=cudart
ollama>
ollama>   #$ _HERE_=/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin
ollama>
ollama>   #$ _THERE_=/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin
ollama>
ollama>   #$ _TARGET_SIZE_=
ollama>
ollama>   #$ _TARGET_DIR_=
ollama>
ollama>   #$ _TARGET_SIZE_=64
ollama>
ollama>   #$ TOP=/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/..
ollama>
ollama>   #$
ollama>   NVVMIR_LIBRARY_DIR=/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/../nvvm/libdevice
ollama>
ollama>
ollama>   #$
ollama>   LD_LIBRARY_PATH=/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/../lib:
ollama>
ollama>
ollama>   #$
ollama>   PATH=/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/../nvvm/bin:/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin:/nix/store/mzg3cka0bb
r5jq96ysymwziw74fnk22m-go-1.22.1/share/go/bin:/nix/store/mzg3cka0bbr5jq96ysymwziw74fnk22m-go-1.22.1/bin:/nix/store/mbw6qprvn80p8k7z3lq44hnz6b332xf6-cmake-3.28.3/bin:/nix/
store/43n2dxi419b1jgv2bq1xg7y6n8k66xpb-patchelf-0.15.0/bin:/nix/store/g2snq446gf970679qn9dknsvyc7ldmmj-gcc-wrapper-12.3.0/bin:/nix/store/b7d36dp6qglv9vs3qn0wcdprhylr8npr-
gcc-12.3.0/bin:/nix/store/fpv9yd0mwm1lyph02j9bdyxbjz8vaxzy-glibc-2.39-5-bin/bin:/nix/store/ifzwv2xqwdnv1gz87rxkizi67py5p3vj-coreutils-9.4/bin:/nix/store/li6hvanskij8agwk8
vcjlwisvbjpyi0c-binutils-wrapper-2.41/bin:/nix/store/hqvni28zpibl6jsqqimcvng6h6qm58xy-binutils-2.41/bin:/nix/store/ifzwv2xqwdnv1gz87rxkizi67py5p3vj-coreutils-9.4/bin:/nix
/store/jg2i1qj5js5ckzvfq5b9p9j1gcv69750-findutils-4.9.0/bin:/nix/store/ksyif7fp4p0baafvdm5szd375s7qf6cz-diffutils-3.10/bin:/nix/store/237dff1igc3v09p9r23a37yw8dr04bv6-gnu
sed-4.9/bin:/nix/store/avqi5nnx7qydr078ssgifc2hgzqipqgx-gnugrep-3.11/bin:/nix/store/2s1h0q7xhk33nsw90l0g53bxax0d19ih-gawk-5.2.2/bin:/nix/store/xwvgwr85gghbl66kj30sp19p2y1
cypqv-gnutar-1.35/bin:/nix/store/vnfi66rm1dykn8gwwy5i4f48bhbk9x6x-gzip-1.13/bin:/nix/store/3bdjxil2cz8dkd6l5sf4qalla762m99d-bzip2-1.0.8-bin/bin:/nix/store/pg3wkw0as88676i
b4ckbibrfwy5c5vzg-gnumake-4.4.1/bin:/nix/store/a1s263pmsci9zykm5xcdf7x9rv26w6d5-bash-5.2p26/bin:/nix/store/bf9rz7vvxannxp15wfi7x0bx5b68ky7z-patch-2.7.6/bin:/nix/store/jdg
5lv2j7d765h8pn73qz72lilxrcb5x-xz-5.4.6-bin/bin:/nix/store/j3f0hag67n4piz2avqgf07gja8a2ifn6-file-5.45/bin
ollama>
ollama>
ollama>   #$
ollama>   INCLUDES="-I/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/..//include"
ollama>
ollama>
ollama>   #$ LIBRARIES=
ollama>   "-L/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/..//lib64/stubs"
ollama>   "-L/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/..//lib64"
ollama>
ollama>   #$ CUDAFE_FLAGS=
ollama>
ollama>   #$ PTXAS_FLAGS=
ollama>
ollama>   #$ rm tmp/a_dlink.reg.c
ollama>
ollama>   #$ gcc -D__CUDA_ARCH_LIST__=520 -E -x c++ -D__CUDACC__ -D__NVCC__
ollama>   "-I/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/..//include"
ollama>   -D__CUDACC_VER_MAJOR__=12 -D__CUDACC_VER_MINOR__=2
ollama>   -D__CUDACC_VER_BUILD__=140 -D__CUDA_API_VER_MAJOR__=12
ollama>   -D__CUDA_API_VER_MINOR__=2 -D__NVCC_DIAG_PRAGMA_SUPPORT__=1 -include
ollama>   "cuda_runtime.h" -m64 "CMakeCUDACompilerId.cu" -o
ollama>   "tmp/CMakeCUDACompilerId.cpp4.ii"
ollama>
ollama>   #$ cudafe++ --c++17 --gnu_version=120300 --display_error_number
ollama>   --orig_src_file_name "CMakeCUDACompilerId.cu" --orig_src_path_name
ollama>   "/build/source/llm/llama.cpp/build/linux/x86_64/cuda_v12/CMakeFiles/3.28.3/CompilerIdCUDA/CMakeCUDACompilerId.cu"
ollama>   --allow_managed --m64 --parse_templates --gen_c_file_name
ollama>   "tmp/CMakeCUDACompilerId.cudafe1.cpp" --stub_file_name
ollama>   "CMakeCUDACompilerId.cudafe1.stub.c" --gen_module_id_file
ollama>   --module_id_file_name "tmp/CMakeCUDACompilerId.module_id"
ollama>   "tmp/CMakeCUDACompilerId.cpp4.ii"
ollama>
ollama>   #$ gcc -D__CUDA_ARCH__=520 -D__CUDA_ARCH_LIST__=520 -E -x c++
ollama>   -DCUDA_DOUBLE_MATH_FUNCTIONS -D__CUDACC__ -D__NVCC__
ollama>   "-I/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/..//include"
ollama>   -D__CUDACC_VER_MAJOR__=12 -D__CUDACC_VER_MINOR__=2
ollama>   -D__CUDACC_VER_BUILD__=140 -D__CUDA_API_VER_MAJOR__=12
ollama>   -D__CUDA_API_VER_MINOR__=2 -D__NVCC_DIAG_PRAGMA_SUPPORT__=1 -include
ollama>   "cuda_runtime.h" -m64 "CMakeCUDACompilerId.cu" -o
ollama>   "tmp/CMakeCUDACompilerId.cpp1.ii"
ollama>
ollama>   #$ cicc --c++17 --gnu_version=120300 --display_error_number
ollama>   --orig_src_file_name "CMakeCUDACompilerId.cu" --orig_src_path_name
ollama>   "/build/source/llm/llama.cpp/build/linux/x86_64/cuda_v12/CMakeFiles/3.28.3/CompilerIdCUDA/CMakeCUDACompilerId.cu"
ollama>   --allow_managed -arch compute_52 -m64 --no-version-ident -ftz=0 -prec_div=1
ollama>   -prec_sqrt=1 -fmad=1 --include_file_name "CMakeCUDACompilerId.fatbin.c"
ollama>   -tused --module_id_file_name "tmp/CMakeCUDACompilerId.module_id"
ollama>   --gen_c_file_name "tmp/CMakeCUDACompilerId.cudafe1.c" --stub_file_name
ollama>   "tmp/CMakeCUDACompilerId.cudafe1.stub.c" --gen_device_file_name
ollama>   "tmp/CMakeCUDACompilerId.cudafe1.gpu" "tmp/CMakeCUDACompilerId.cpp1.ii" -o
ollama>   "tmp/CMakeCUDACompilerId.ptx"
ollama>
ollama>   #$ ptxas -arch=sm_52 -m64 "tmp/CMakeCUDACompilerId.ptx" -o
ollama>   "tmp/CMakeCUDACompilerId.sm_52.cubin"
ollama>
ollama>   #$ fatbinary --create="tmp/CMakeCUDACompilerId.fatbin" -64
ollama>   --cicc-cmdline="-ftz=0 -prec_div=1 -prec_sqrt=1 -fmad=1 "
ollama>   "--image3=kind=elf,sm=52,file=tmp/CMakeCUDACompilerId.sm_52.cubin"
ollama>   "--image3=kind=ptx,sm=52,file=tmp/CMakeCUDACompilerId.ptx"
ollama>   --embedded-fatbin="tmp/CMakeCUDACompilerId.fatbin.c"
ollama>
ollama>   #$ gcc -D__CUDA_ARCH__=520 -D__CUDA_ARCH_LIST__=520 -c -x c++
ollama>   -DCUDA_DOUBLE_MATH_FUNCTIONS
ollama>   "-I/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/..//include"
ollama>   -m64 "tmp/CMakeCUDACompilerId.cudafe1.cpp" -o "tmp/CMakeCUDACompilerId.o"
ollama>
ollama>   #$ nvlink -m64 --arch=sm_52 --register-link-binaries="tmp/a_dlink.reg.c"
ollama>   "-L/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/..//lib64/stubs"
ollama>   "-L/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/..//lib64"
ollama>   -cpu-arch=X86_64 "tmp/CMakeCUDACompilerId.o" -lcudadevrt -o
ollama>   "tmp/a_dlink.sm_52.cubin" --host-ccbin "gcc"
ollama>
ollama>   #$ fatbinary --create="tmp/a_dlink.fatbin" -64 --cicc-cmdline="-ftz=0
ollama>   -prec_div=1 -prec_sqrt=1 -fmad=1 " -link
ollama>   "--image3=kind=elf,sm=52,file=tmp/a_dlink.sm_52.cubin"
ollama>   --embedded-fatbin="tmp/a_dlink.fatbin.c"
ollama>
ollama>   #$ gcc -D__CUDA_ARCH_LIST__=520 -c -x c++
ollama>   -DFATBINFILE="\"tmp/a_dlink.fatbin.c\""
ollama>   -DREGISTERLINKBINARYFILE="\"tmp/a_dlink.reg.c\"" -I.
ollama>   -D__NV_EXTRA_INITIALIZATION= -D__NV_EXTRA_FINALIZATION=
ollama>   -D__CUDA_INCLUDE_COMPILER_INTERNAL_HEADERS__
ollama>   "-I/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/..//include"
ollama>   -D__CUDACC_VER_MAJOR__=12 -D__CUDACC_VER_MINOR__=2
ollama>   -D__CUDACC_VER_BUILD__=140 -D__CUDA_API_VER_MAJOR__=12
ollama>   -D__CUDA_API_VER_MINOR__=2 -D__NVCC_DIAG_PRAGMA_SUPPORT__=1 -m64
ollama>   "/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/crt/link.stub"
ollama>   -o "tmp/a_dlink.o"
ollama>
ollama>   #$ g++ -D__CUDA_ARCH_LIST__=520 -m64 -Wl,--start-group "tmp/a_dlink.o"
ollama>   "tmp/CMakeCUDACompilerId.o"
ollama>   "-L/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/..//lib64/stubs"
ollama>   "-L/nix/store/3ddhlzkjcysyywnhnzmi16s9spr5d3jc-cuda-toolkit/bin/..//lib64"
ollama>   -lcudadevrt -lcudart_static -lrt -lpthread -ldl -Wl,--end-group -o "a.out"
ollama>
ollama>   /nix/store/hqvni28zpibl6jsqqimcvng6h6qm58xy-binutils-2.41/bin/ld: cannot
ollama>   find -lcudart_static: No such file or directory
ollama>
ollama>   collect2: error: ld returned 1 exit status
ollama>
ollama>   # --error 0x1 --
ollama>
ollama>
ollama>
ollama>
ollama>
ollama> Call Stack (most recent call first):
ollama>   /nix/store/mbw6qprvn80p8k7z3lq44hnz6b332xf6-cmake-3.28.3/share/cmake-3.28/Modules/CMakeDetermineCompilerId.cmake:8 (CMAKE_DETERMINE_COMPILER_ID_BUILD)
ollama>   /nix/store/mbw6qprvn80p8k7z3lq44hnz6b332xf6-cmake-3.28.3/share/cmake-3.28/Modules/CMakeDetermineCompilerId.cmake:53 (__determine_compiler_id_test)
ollama>   /nix/store/mbw6qprvn80p8k7z3lq44hnz6b332xf6-cmake-3.28.3/share/cmake-3.28/Modules/CMakeDetermineCUDACompiler.cmake:135 (CMAKE_DETERMINE_COMPILER_ID)
ollama>   CMakeLists.txt:369 (enable_language)
ollama>
ollama>
ollama> -- Configuring incomplete, errors occurred!
ollama> llm/generate/generate_linux.go:3: running "bash": exit status 1
ollama> /nix/store/69b30l95dg137z7fp3y67fl8l90rilgl-stdenv-linux/setup: line 131: pop_var_context: head of shell_variables not a function context
error: builder for '/nix/store/6p9fn8ckfp3f341050w2j6bgwxm6s8wp-ollama-0.1.30.drv' failed with exit code 1;
       last 10 log lines:
       > Call Stack (most recent call first):
       >   /nix/store/mbw6qprvn80p8k7z3lq44hnz6b332xf6-cmake-3.28.3/share/cmake-3.28/Modules/CMakeDetermineCompilerId.cmake:8 (CMAKE_DETERMINE_COMPILER_ID_BUILD)
       >   /nix/store/mbw6qprvn80p8k7z3lq44hnz6b332xf6-cmake-3.28.3/share/cmake-3.28/Modules/CMakeDetermineCompilerId.cmake:53 (__determine_compiler_id_test)
       >   /nix/store/mbw6qprvn80p8k7z3lq44hnz6b332xf6-cmake-3.28.3/share/cmake-3.28/Modules/CMakeDetermineCUDACompiler.cmake:135 (CMAKE_DETERMINE_COMPILER_ID)
       >   CMakeLists.txt:369 (enable_language)
       >
       >
       > -- Configuring incomplete, errors occurred!
       > llm/generate/generate_linux.go:3: running "bash": exit status 1
       > /nix/store/69b30l95dg137z7fp3y67fl8l90rilgl-stdenv-linux/setup: line 131: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/6p9fn8ckfp3f341050w2j6bgwxm6s8wp-ollama-0.1.30.drv'.
error: 1 dependencies of derivation '/nix/store/srzi5jvsbg4ppjny5jgzv4a6i94jh21w-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/c4lnd14y5iljy5qs8fxi6q9w9v2wj2lq-unit-ollama.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/a03wvjnvf22y5gnmk83s6z3v0318ajvh-nixos-system-firefly-24.05.20240408.4cba8b5.drv' failed to build
```

### Additional context

Add any other context about the problem here.

### Notify maintainers
@abysssol @onny
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.4-cachyos, NixOS, 24.05 (Uakari), 24.05.20240408.4cba8b5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/450afzqlzzgw6wnyc3dwysf3i5yxyqkr-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

Blender with cuda is also broekn maybe it is linked ?

I enabled cuda with this `(blender.override { cudaSupport = true; })`

<details>

<summary>build log</summary>

```
Sourcing setup-cuda-hook
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/4xv426wpwyj54nrfmpyff7218ylm8c1a-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
applying patch /nix/store/922qaxfmdgjhk64sx6gqx2iqdxkn3401-cuda.patch
patching file devices/CMakeLists.txt
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
Executing setupCUDAToolkitCompilers
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/9fy4109a1zgwnk9gwy8fq1kynd7629mp-openimagedenoise-2.2.2/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/9fy4109a1zgwnk9gwy8fq1kynd7629mp-openimagedenoise-2.2.2/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/9fy4109a1zgwnk9gwy8fq1kynd7629mp-openimagedenoise-2.2.2/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/9fy4109a1zgwnk9gwy8fq1kynd7629mp-openimagedenoise-2.2.2/share/doc/OpenImageDenoise -DCMAKE_INSTALL_INFODIR=/nix/store/9fy4109a1zgwnk9gwy8fq1kynd7629mp-openimagedenoise-2.2.2/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/9fy4109a1zgwnk9gwy8fq1kynd7629mp-openimagedenoise-2.2.2/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/9fy4109a1zgwnk9gwy8fq1kynd7629mp-openimagedenoise-2.2.2/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/9fy4109a1zgwnk9gwy8fq1kynd7629mp-openimagedenoise-2.2.2/include -DCMAKE_INSTALL_SBINDIR=/nix/store/9fy4109a1zgwnk9gwy8fq1kynd7629mp-openimagedenoise-2.2.2/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/9fy4109a1zgwnk9gwy8fq1kynd7629mp-openimagedenoise-2.2.2/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/9fy4109a1zgwnk9gwy8fq1kynd7629mp-openimagedenoise-2.2.2/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/ac1hb5dc2z4biwgy8mjrhlifffkkrvdq-gcc-wrapper-13.2.0/bin/strip -DCMAKE_RANLIB=/nix/store/ac1hb5dc2z4biwgy8mjrhlifffkkrvdq-gcc-wrapper-13.2.0/bin/ranlib -DCMAKE_AR=/nix/store/ac1hb5dc2z4biwgy8mjrhlifffkkrvdq-gcc-wrapper-13.2.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/9fy4109a1zgwnk9gwy8fq1kynd7629mp-openimagedenoise-2.2.2 -DOIDN_DEVICE_CUDA:BOOL=TRUE -DTBB_INCLUDE_DIR:STRING=/nix/store/zw6p7cv8ya93c8zjmwyr1bh4pml77yil-tbb-2020.3-dev/include -DTBB_ROOT:STRING=/nix/store/n1ixg77c3xshkng1d5nknhpknx7y7ffc-tbb-2020.3 -DCUDAToolkit_INCLUDE_DIR=/nix/store/m784c2rsx1gz5aj3lzy32k4rd5m1iv5g-cuda_cudart-12.2.140-dev/include;/nix/store/6wrzb2nldjw8rm9q2yvlwk53wb2hl2dn-cuda_cccl-12.2.140-dev/include;/nix/store/677gn3v8lga776iv3ncqpw94l2f3xvq8-cuda_nvcc-12.2.140-dev/include -DCUDAToolkit_ROOT=/nix/store/m784c2rsx1gz5aj3lzy32k4rd5m1iv5g-cuda_cudart-12.2.140-dev;/nix/store/6wrzb2nldjw8rm9q2yvlwk53wb2hl2dn-cuda_cccl-12.2.140-dev;/nix/store/677gn3v8lga776iv3ncqpw94l2f3xvq8-cuda_nvcc-12.2.140-dev -DCUDA_HOST_COMPILER=/nix/store/63z0cql37dljbszd4fbwrpnhskc00kp4-gcc-wrapper-12.3.0/bin/c++ -DCMAKE_CUDA_HOST_COMPILER=/nix/store/63z0cql37dljbszd4fbwrpnhskc00kp4-gcc-wrapper-12.3.0/bin/c++ 
-- The C compiler identification is GNU 13.2.0
-- The CXX compiler identification is GNU 13.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/ac1hb5dc2z4biwgy8mjrhlifffkkrvdq-gcc-wrapper-13.2.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/ac1hb5dc2z4biwgy8mjrhlifffkkrvdq-gcc-wrapper-13.2.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Python: /nix/store/gd3shnza1i50zn8zs04fa729ribr88m9-python3-3.11.8/bin/python3.11 (found version "3.11.8") found components: Interpreter 
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Could NOT find TBB (missing: TBB_DIR)
-- Looking for TBB components tbb; ()
-- Found TBB version 2020.3 at /nix/store/n1ixg77c3xshkng1d5nknhpknx7y7ffc-tbb-2020.3
-- Found Intel SPMD Compiler (ISPC): /nix/store/376iwj0v3d503xaahhysf9vzdijylyhb-ispc-1.23.0/bin/ispc
-- Configuring done (2.9s)
-- Generating done (0.1s)
[33mCMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025
    CUDAToolkit_INCLUDE_DIR
    CUDA_HOST_COMPILER

[0m
-- Build files have been written to: /build/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
build flags: -j12 SHELL=/nix/store/a1s263pmsci9zykm5xcdf7x9rv26w6d5-bash-5.2p26/bin/bash
[  1%] [34m[1mGenerating CXX source files from blob ../weights/rt_alb.tza[0m
[  2%] [34m[1mGenerating CXX source files from blob ../weights/rtlightmap_hdr.tza[0m
[  3%] [34m[1mGenerating CXX source files from blob ../weights/rt_hdr.tza[0m
[  4%] [34m[1mGenerating CXX source files from blob ../weights/rt_hdr_alb.tza[0m
[  5%] [34m[1mGenerating CXX source files from blob ../weights/rt_ldr_calb_cnrm.tza[0m
[  7%] [34m[1mGenerating CXX source files from blob ../weights/rt_hdr_alb_nrm.tza[0m
[  8%] [34m[1mGenerating CXX source files from blob ../weights/rt_ldr_alb.tza[0m
[  9%] [34m[1mGenerating CXX source files from blob ../weights/rt_ldr_alb_nrm.tza[0m
[ 10%] [34m[1mGenerating CXX source files from blob ../weights/rt_hdr_calb_cnrm.tza[0m
[ 10%] [34m[1mGenerating CXX source files from blob ../weights/rt_ldr.tza[0m
[ 11%] [34m[1mGenerating CXX source files from blob ../weights/rt_nrm.tza[0m
[ 12%] [32mBuilding CXX object common/CMakeFiles/OpenImageDenoise_common.dir/common.cpp.o[0m
[ 13%] [32mBuilding CXX object common/CMakeFiles/OpenImageDenoise_common.dir/half.cpp.o[0m
[ 14%] [32mBuilding CXX object common/CMakeFiles/OpenImageDenoise_common.dir/oidn_utils.cpp.o[0m
[ 15%] [34m[1mGenerating CXX source files from blob ../weights/rtlightmap_dir.tza[0m
[ 16%] [32mBuilding CXX object common/CMakeFiles/OpenImageDenoise_common.dir/platform.cpp.o[0m
[ 16%] Built target OpenImageDenoise_common
[ 19%] [32mBuilding CXX object CMakeFiles/OpenImageDenoise_weights.dir/weights/rt_hdr_calb_cnrm.cpp.o[0m
[ 19%] [32mBuilding CXX object CMakeFiles/OpenImageDenoise_weights.dir/weights/rt_ldr.cpp.o[0m
[ 20%] [32mBuilding CXX object CMakeFiles/OpenImageDenoise_weights.dir/weights/rt_hdr_alb_nrm.cpp.o[0m
[ 20%] [32mBuilding CXX object CMakeFiles/OpenImageDenoise_weights.dir/weights/rt_alb.cpp.o[0m
[ 21%] [32mBuilding CXX object CMakeFiles/OpenImageDenoise_weights.dir/weights/rt_ldr_alb_nrm.cpp.o[0m
[ 23%] [32mBuilding CXX object CMakeFiles/OpenImageDenoise_weights.dir/weights/rt_hdr_alb.cpp.o[0m
[ 23%] [32mBuilding CXX object CMakeFiles/OpenImageDenoise_weights.dir/weights/rt_hdr.cpp.o[0m
[ 24%] [32mBuilding CXX object CMakeFiles/OpenImageDenoise_weights.dir/weights/rt_ldr_alb.cpp.o[0m
[ 25%] [32mBuilding CXX object CMakeFiles/OpenImageDenoise_weights.dir/weights/rtlightmap_dir.cpp.o[0m
[ 26%] [32mBuilding CXX object CMakeFiles/OpenImageDenoise_weights.dir/weights/rtlightmap_hdr.cpp.o[0m
[ 27%] [32mBuilding CXX object CMakeFiles/OpenImageDenoise_weights.dir/weights/rt_ldr_calb_cnrm.cpp.o[0m
[ 28%] [32mBuilding CXX object CMakeFiles/OpenImageDenoise_weights.dir/weights/rt_nrm.cpp.o[0m
[ 28%] Built target OpenImageDenoise_weights
[ 29%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/buffer.cpp.o[0m
[ 30%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/concat_conv_chw.cpp.o[0m
[ 31%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/concat_conv.cpp.o[0m
[ 32%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/color.cpp.o[0m
[ 33%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/context.cpp.o[0m
[ 34%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/arena.cpp.o[0m
[ 36%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/arena_planner.cpp.o[0m
[ 36%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/concat_conv_hwc.cpp.o[0m
[ 37%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/conv.cpp.o[0m
[ 38%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/exception.cpp.o[0m
[ 39%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/device.cpp.o[0m
[ 40%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/engine.cpp.o[0m
[ 41%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/filter.cpp.o[0m
[ 42%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/graph.cpp.o[0m
[ 43%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/heap.cpp.o[0m
[ 44%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/image.cpp.o[0m
[ 45%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/input_process.cpp.o[0m
[ 46%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/module.cpp.o[0m
[ 47%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/output_process.cpp.o[0m
[ 48%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/pool.cpp.o[0m
[ 49%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/progress.cpp.o[0m
[ 50%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/rt_filter.cpp.o[0m
[ 51%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/rtlightmap_filter.cpp.o[0m
[ 52%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/subdevice.cpp.o[0m
[ 53%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/tensor.cpp.o[0m
[ 54%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/tensor_reorder.cpp.o[0m
[ 55%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/thread.cpp.o[0m
[ 56%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/tza.cpp.o[0m
[ 57%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/unet_filter.cpp.o[0m
[ 58%] [32mBuilding CXX object core/CMakeFiles/OpenImageDenoise_core.dir/upsample.cpp.o[0m
[ 59%] [32m[1mLinking CXX shared library ../libOpenImageDenoise_core.so[0m
[ 59%] Built target OpenImageDenoise_core
[ 60%] [34m[1mBuilding ISPC object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_autoexposure.dev.o[0m
[ 61%] [34m[1mBuilding ISPC object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/color.dev.o[0m
[ 62%] [34m[1mBuilding ISPC object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_conv.dev.o[0m
[ 63%] [34m[1mBuilding ISPC object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_image_copy.dev.o[0m
[ 64%] [34m[1mBuilding ISPC object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_upsample.dev.o[0m
[ 66%] [34m[1mBuilding ISPC object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_output_process.dev.o[0m
[ 66%] [34m[1mBuilding ISPC object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/platform.dev.o[0m
[ 67%] [34m[1mBuilding ISPC object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_pool.dev.o[0m
[ 68%] [34m[1mBuilding ISPC object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_input_process.dev.o[0m
[ 68%] [32mBuilding CXX object api/CMakeFiles/OpenImageDenoise.dir/api.cpp.o[0m
[ 69%] [34m[1mCreating directories for 'OpenImageDenoise_device_cuda'[0m
[ 70%] [34m[1mNo download step for 'OpenImageDenoise_device_cuda'[0m
[ 71%] [34m[1mNo update step for 'OpenImageDenoise_device_cuda'[0m
[ 72%] [34m[1mNo patch step for 'OpenImageDenoise_device_cuda'[0m
[ 73%] [34m[1mPerforming configure step for 'OpenImageDenoise_device_cuda'[0m
loading initial cache file /build/source/build/devices/cuda/tmp/OpenImageDenoise_device_cuda-cache-Release.cmake
-- The CXX compiler identification is GNU 13.2.0
[31mCMake Error at /nix/store/mbw6qprvn80p8k7z3lq44hnz6b332xf6-cmake-3.28.3/share/cmake-3.28/Modules/Internal/CMakeCUDAFindToolkit.cmake:37 (message):
  Could not find nvcc executable in path specified by
  CUDAToolkit_ROOT=/nix/store/m784c2rsx1gz5aj3lzy32k4rd5m1iv5g-cuda_cudart-12.2.140-dev;/nix/store/6wrzb2nldjw8rm9q2yvlwk53wb2hl2dn-cuda_cccl-12.2.140-dev;/nix/store/677gn3v8lga776iv3ncqpw94l2f3xvq8-cuda_nvcc-12.2.140-dev
Call Stack (most recent call first):
  /nix/store/mbw6qprvn80p8k7z3lq44hnz6b332xf6-cmake-3.28.3/share/cmake-3.28/Modules/CMakeDetermineCUDACompiler.cmake:89 (cmake_cuda_find_toolkit)
  CMakeLists.txt:17 (project)

[0m
-- Configuring incomplete, errors occurred!
make[2]: *** [devices/CMakeFiles/OpenImageDenoise_device_cuda.dir/build.make:92: devices/cuda/stamp/OpenImageDenoise_device_cuda-configure] Error 1
make[1]: *** [CMakeFiles/Makefile2:316: devices/CMakeFiles/OpenImageDenoise_device_cuda.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 74%] [32m[1mLinking CXX shared library ../libOpenImageDenoise.so[0m
[ 74%] Built target OpenImageDenoise
[ 77%] [32mBuilding CXX object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_engine.cpp.o[0m
[ 77%] [32mBuilding CXX object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_device.cpp.o[0m
[ 77%] [32mBuilding CXX object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_common.cpp.o[0m
[ 78%] [32mBuilding CXX object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_pool.cpp.o[0m
[ 79%] [32mBuilding CXX object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_image_copy.cpp.o[0m
[ 80%] [32mBuilding CXX object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_autoexposure.cpp.o[0m
[ 82%] [32mBuilding CXX object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_output_process.cpp.o[0m
[ 82%] [32mBuilding CXX object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_upsample.cpp.o[0m
[ 82%] [32mBuilding CXX object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_input_process.cpp.o[0m
[ 83%] [32mBuilding CXX object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_conv.cpp.o[0m
[ 84%] [32mBuilding CXX object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/tasking.cpp.o[0m
[ 85%] [32mBuilding CXX object devices/cpu/CMakeFiles/OpenImageDenoise_device_cpu.dir/cpu_module.cpp.o[0m
[ 86%] [32m[1mLinking CXX shared library ../../libOpenImageDenoise_device_cpu.so[0m
[ 86%] Built target OpenImageDenoise_device_cpu
make: *** [Makefile:156: all] Error 2
```
</details>

> It is possible to enable CUDA for you system at `flakes.nix` (at config, or when importing `nixpkgs` as `config`) or as a usual configuration:
> 
> > nixpkgs.config.cudaSupport = true;
> 
> (This won't fix Ollama. But it enables CUDA for testing.)
> 
> I was able to reproduce this problem here: https://termbin.com/b7r4
> 
> > cannot find -lcudart_static: No such file or directory

Here is my modifications and I juste don't have cuda detected in blender using this
```patch
diff --git a/flake.nix b/flake.nix
index 4dcb29a..3654a34 100644
--- a/flake.nix
+++ b/flake.nix
@@ -20,7 +20,9 @@
 
   inputs = {
 # Official NixOS package source, using nixos-unstable branch here
-    nixpkgs.url = "nixpkgs/nixos-unstable";
+    nixpkgs = {
+      url = "nixpkgs/nixos-unstable";
+    };
     nur.url = github:nix-community/NUR;
 
     lanzaboote = {
@@ -45,7 +47,10 @@
   outputs = { self, nixpkgs, home-manager, flatpaks, plasma-manager, nur, lanzaboote, ... }@inputs:
     let
     system = "x86_64-linux";
-    pkgs = import nixpkgs { inherit system; };
+    pkgs = import nixpkgs { 
+    	inherit system;  
+	config.cudaSupport = true;
+    };
   in {
     nixosConfigurations = {
       nixos-eymeric = nixpkgs.lib.nixosSystem {
diff --git a/home.nix b/home.nix
index 70e80f7..57f853b 100644
--- a/home.nix
+++ b/home.nix
@@ -165,7 +165,8 @@
       git-crypt
       delta
       skrooge
-      (blender.override { cudaSupport = true; })
+      blender
+      #(blender.override { cudaSupport = true; })
       skypeforlinux
       floorp
       fh
```

--- a/pkgs/tools/misc/ollama/default.nix
+++ b/pkgs/tools/misc/ollama/default.nix
@@ -86,6 +86,7 @@ let
     paths = [
       cudaPackages.cudatoolkit
       cudaPackages.cuda_cudart
+      cudaPackages.cuda_cudart.static
     ];
     postBuild = ''
       rm "$out/lib64"

